### PR TITLE
SS4 upgrade

### LINKED ISSUE #14: SilverStripe 4 support
It would be great to get SS4 support going on this so that silvershop can be upgraded.

### PR DESCRIPTION
**Pull request by fspringveldt**
Fixes #14 

* Ran upgrader tool
* Ran PHP linter
* Upgraded unit tests
* Updated composer config
* Updated readme
* Fixed typo in template file
* Added build badge to readme

### CODE REVIEW

File: src/ListSorter.php [hunk 1]
```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace SilverStripe\ListSorter;
```
Comment (sminnee) [inline src/ListSorter.php]: This should be `SilverShop\ListSorter`

File: composer.json [hunk 2]
```diff
@@ -10,10 +10,17 @@
...
-        "silverstripe/cms": "^3.0"
+        "silverstripe/cms": "~4.0"
     },
 	 "extra": {
         "installer-name": "listsorter"
-    }
+    },
+    "autoload": {
+        "psr-4": {
+            "SilverStripe\\ListSorter\\": "src/",
```
Comment (sminnee) [inline composer.json]: SilverShop\ListSorter on both these items.

File: composer.json [hunk 3]
```diff
@@ -10,10 +10,17 @@
         }
     ],
     "require": {
-        "silverstripe/framework": "^3.0",
-        "silverstripe/cms": "^3.0"
+        "silverstripe/cms": "~4.0"
     },
 	 "extra": {
```
Comment (sminnee) [inline composer.json]: Remove this.

**Comment by sminnee** [comment]
Hi @fspringveldt - looking like a good start but as well as the test failures I had a few notes. In comments above.

 * Namespace should be SilverShop\ListSorter
 * Use type: silverstripe-vendormodule and remove installer-name

File: codecov.yml [hunk 4]
```diff
@@ -0,0 +1 @@
+comment: false
```
Comment (sminnee) [inline codecov.yml]: What's the benefit of this?

**Comment by bummzack** [comment]
Thanks a lot for this!

**Pull request merged**